nixos/modules/services/networking/openconnect.nix

```diff
@@ -0,0 +1,138 @@
+{ config, lib, options, pkgs, ... }: with lib;
+let
+  cfg = config.networking.openconnect;
+  openconnect = cfg.package;
+  pkcs11 = types.strMatching "pkcs11:.+" // { name = "pkcs11"; description = "PKCS#11 URI"; };
+  interfaceOptions = {
+    options = {
+      gateway = mkOption {
+        description = "Gateway server to connect to.";
+        example = "gateway.example.com";
+        type = types.str;
+      };
+
+      protocol = mkOption {
+        description = ''
+          Protocol to use.
+        '';
+        example = "anyconnect";
+        type = types.enum [ "anyconnect" "array" "nc" "pulse" "gp" "f5" "fortinet" ];
+      };
+
+      user = mkOption {
+        description = "Username to authenticate with.";
+        example = "example-user";
+        type = types.nullOr types.str;
+      };
+
+      # Note: It does not make sense to provide a way to declaratively
+      # set an authentication cookie, because they have to be requested
+      # for every new connection and would only work once.
+      password = mkOption {
```

Comment:
Please remove this option.

Reply:
Could you provide your reasoning behind this?

Reply:
[This is bad for security](https://github.com/NixOS/nixpkgs/issues/24288). We shouldn't accept new code into NixOS that does this unnecessarily.

Reply:
Doesn't the warning in the description and the `passwordFile` option right below exactly implement the proposal from the issue you linked? As far as I understand, the goal is not to eliminate all options that store secrets in the Nix store (as there might be use cases where this is not a problem), but rather to provide a way to avoid doing that.

Reply:
> but rather to provide a way to avoid doing that.

Yes. Like so: `pkgs.writeText "bad-idea" "not-a-secure-password"`. `password` as an option becomes redundant - and - encourages users into bad behavior.

Reply:
I thought that it would be convenient to provide the `password` option to users who don't care about secrets in the Nix store (like on private machines), but ultimately that's a design choice. I'll remove the option.

Reply:
> I thought that it would be convenient
My point exactly.

> I'll remove the option.

Thanks.